test/anchor_sup_tests.erl

```diff
@@ -0,0 +1,57 @@
+-module(anchor_sup_tests).
+-include_lib("anchor/include/anchor.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+anchor_sup_creates_all_pools_test_() ->
+    {setup,
+     fun() ->
+             application:set_env(anchor, pools, [{foo, []}, {bar, []}]),
+             application:ensure_all_started(shackle),
+             anchor_sup:start_link(),
+             ok
+     end,
+     fun(_) ->
+             shackle_pool:stop(foo),
+             shackle_pool:stop(bar),
+             ok
+     end,
+     [?_assertEqual(2 * ?DEFAULT_POOL_SIZE,
+                           length(supervisor:which_children(shackle_sup)))]}.
+
+anchor_uses_config_from_env_test_() ->
+    {setup,
+     fun() ->
+             meck:new(shackle_pool),
+             meck:expect(shackle_pool, start, fun(_, _, _, _) -> ok end),
+             ok
+     end,
+     fun(_) ->
+             meck:unload(shackle_pool)
+     end,
+     [fun() ->
+              MemcachedConfig = [{ip, 1},
+                                 {port, 4},
+                                 {reconnect, 5},
+                                 {reconnect_time_max, 6},
+                                 {reconnect_time_min, 7},
+                                 {socket_options, 8}],
+              PoolConfig = [{backlog_size, 0},
+                            {pool_size, 2},
+                            {pool_strategy, 3}],
+              application:set_env(anchor,
+                                  pools,
+                                  [{foo, [{backlog_size, 0},
+                                          {ip, 1},
+                                          {pool_size, 2},
+                                          {pool_strategy, 3},
+                                          {port, 4},
+                                          {reconnect, 5},
+                                          {reconnect_time_max, 6},
+                                          {reconnect_time_min, 7},
+                                          {socket_options, 8}]}]),
+              anchor_sup:init([]),
+              StartArgs = [foo, anchor_client, MemcachedConfig, PoolConfig],
+              ?assert(meck:called(shackle_pool,
```

Comment:
The merits are debatable, but I think you can get rid of `meck` here as well by using `supervisor:get_childspec/2`; the options you're asserting do exist there. One problem I could see is that it introduces a coupling between this test code and the order/format of the arguments passed to the child process, but maybe that isn't such a bad thing?

Reply:
By getting rid of meck entirely in favor of `supervisor:get_childspec/2`, I'd be calling the real `shackle_pool:start/4`, which takes care of launching the child processes. By doing so, Shackle would try to reach a memcached server, which is not what I'm trying to achieve in this unit test.

But I do agree that, generally speaking, `supervisor:get_childspec/2` is a good option when doing integration testing, as it allows to test the behavior (i.e. the pools have been created with correct parameters) and not the implementation details (i.e. we call `shackle_pool:start`).

Reply:
Also while Travis allows to launch a memcached service through .travis-ci.yml, it is quite limited. We can only launch one memcached instance running on localhost with default parameters and port.

Reply:
I'm pretty sure the child processes will start even if there's no memcached instance to connect to; they will print errors, but they should spawn successfully.